Yoast/YoastSEO.js
# Add synonyms and keyword distribution assessment

## Summary

This PR can be summarized in the following changelog entry:

* Adds an assessment the checks the distribution of the focus keyword in the text.
* Adds a a topic distribution and topic density assessment to use with synonyms

## Relevant technical choices:

* The keyword distribution assessment is added in this PR, but it's only available in Premium.

## Test instructions

This PR can be tested by following these steps:

* Make a beta version with Yoast SEO Premium (trunk), Yoast Components (develop) and YoastSEO.js (this branch). In this beta version test the following functionality:
  * Add a post with at least 100 words, set a keyword and use the keyword at least twice in the text.
  * Add a synonym or several synonyms (separated by commas) into the new synonym input field.
  * Make sure the topic distribution and topic density assessments are shown. Also check whether markings work. When you add the synonym in the text, the topic density should change.
  * Remove the synonyms. Make sure that instead of topic density and topic distribution, keyword density and keyword distribution are shown. 
* Make a beta version with Yoast SEO Free (trunk), Yoast Components (develop) and YoastSEO.js (this branch). Make sure the keyword distribution assessment isn't triggered.
